File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.py
class Solution(object):
    def maxProfit(self, prices):
        """
        :type prices: List[int]
        :rtype: int
        """
        fin = 0
        curr = 0
        l = len(prices)
        for i in range(l-1):
            profit = prices[i+1] - prices[i]
            curr += profit
            if(curr < 0):
                curr = 0
            else:
                if(curr > fin):
                    fin = curr
        return fin

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
        int maxProf = 0;
        int currProf = 0;
        for(int i = 0;i<prices.length-1;i++){
            int diff = prices[i+1]-prices[i];
            currProf+=diff;
            if(currProf<0){
                currProf = 0;
            }else{
                maxProf = Math.max(maxProf,currProf);
            }
        }
        return maxProf;
    }
}